### DISCUSSION TITLE
Will useAsyncData cache return?

### DISCUSSION THREAD

**Discussion by frezzzen** [nuxt/nuxt#16256] [Questions]
Data caching was so useful, why was it deleted before release?

For example:
I have many APIs that are repeating on certain pages, rather creating my store and custom implementations I could just use useAsyncData and made sure that key is unique, so data will not be fetched if i used it on different place.If I wanted to ensure on visiting page data is refetched I could just use initialCache: false and that was it.

**Comment by Baloche** [comment]
I was about to post the same message ! 
This feature seems essential to me and without it, I am more and more thinking about using an external async state library (👀 @tanstack/vue-query) even if it implies client-only fetching. 😞 

(The related issue, for reference : #8885)

**Reply by Baloche** [reply]
Also, why request deduplication is not `false` by default ?
I have several component on the same page all using the same async data coming from `useFetch('/api/mydata')` and they all trigger a request 🙁

**Reply by Baloche** [reply]
As a workaround, to get back initial cache and no request deduplication I now use this function : 
```ts
function useAsyncDataWithCache(
  key,
  handler,
  { watch: watchSource, ...options } = {}
) {
  const result = useAsyncData(key, handler, { immediate: false, ...options })
  const refresh = result.refresh
  result.refresh = result.execute = (opts) =>
    refresh({ dedupe: false, ...opts })
  if (process.client) result.execute({ _initial: true })
  if (watchSource) watch(watchSource, () => result.refresh())
  return result
}
```

but it feels like a dirty fix 😕

**Reply by frezzzen** [reply]
Yea tried same, it's even dirtier in typescript
